mlflow/store/sqlalchemy_store.py

```diff
@@ -68,6 +91,25 @@ def __init__(self, db_uri, default_artifact_root):
             with self.ManagedSessionMaker() as session:
                 self._create_default_experiment(session)
 
+    @staticmethod
+    def _initialize_tables(engine):
+        _logger.info("Creating initial MLflow database tables...")
+        LegacyBase.metadata.create_all(engine)
+        _upgrade_db(str(engine.url))
+
+    @staticmethod
+    def _verify_schema(engine):
+        mc = MigrationContext.configure(engine.connect())
+        diff = compare_metadata(mc, Base.metadata)
+        if len(diff) > 0:
+            _logger.error("Detected one or more differences between current database schema and "
+                          "desired schema, exiting. Diff:\n %s",
+                          pprint.pprint(diff, indent=2, width=20))
+            raise MlflowException(
+                "Detected out-of-date database schema. Run 'mlflow db upgrade %s' to migrate your "
```

Comment:
Can we explicitly recommend that users back up their database before running this command? I ended up with an MySQL database that was stuck in an intermediate state when attempting the migration as is; I can imagine that it would be very frustrating for users if our migrations caused DB issues and we did not warn them first about creating backups.